### ISSUE TITLE
Filter vulnerabilities found in old versions of the kernel

### ISSUE THREAD

**Issue by mikhailzakhryapin** [aquasecurity/trivy#3764]
Hi,

1. Trivy detects vulnerability in kernel version 1. 
2. I run `apt upgrade` and upgrade kernel to version 2 that address the vulnerability found in step 1. 
3. Now host runs kernel version 2. Trivy shows vulnerability on this host because kernel version 1 is still on the disk. 
4. If I run `apt autoremove`,  kernel version 1 is removed from the disk and Trivy shows that there is no vulnerability on this host 

I would like to have command line option that will hide old version of the kernel appearing in the Trivy report.

**Comment by itaysk** [comment]
We also need to consider how this works with different targets. Scanning kernel vulnerabilities via image scanning doesn't make alot of sense given that containers are abstracted away from the kernel (and trivy is often run as container itself).
Would it make sense to make this a VM target feature? then we can also detect the running kernel with more confidence or have target specific flags (like k8s api-version flag in k8s target)

**Comment by knqyf263** [comment]
First, container images don't show the Linux kernel in the list of installed packages. Trivy doesn't detect kernel vulnerabilities anyway.
This issue is mainly relevant to host scanning like `trivy rootfs` or `trivy vm`. @mikhailzakhryapin Please correct me if I'm wrong.

**Comment by mikhailzakhryapin** [comment]
@knqyf263 That's correct. It relates to scanning host. Sorry that I did not mention it in my first post. We are developing the running kernel detection feature and ready to share the code as soon as we finish  it.

**Comment by knqyf263** [comment]
Great! Looking forward to seeing it.

**Comment by swallace-ciq** [comment]
> @knqyf263 That's correct. It relates to scanning host. Sorry that I did not mention it in my first post. We are developing the running kernel detection feature and ready to share the code as soon as we finish it.

Is there any update on this feature? Definitely interested in it.

**Comment by omerfsen** [comment]
Any update?

**Comment by apdibbo** [comment]
I'd also be very interested in this as a cloud provider

**Comment by freshpr** [comment]
Do you think this will ever be implemented?
I'm getting tired by false positives on a kernel that is not in use but kept as a backup hehe
